src/app/item-review/destiny-tracker.service.ts

```diff
@@ -185,20 +66,20 @@ export function DestinyTrackerService(): DestinyTrackerServiceType {
 
       if (stores[0].isDestiny1()) {
         _bulkFetcher.attachRankings(null,
-                                    stores);
+                                    stores as D1Store[]);
       } else if (stores[0].isDestiny2()) {
         _d2bulkFetcher.attachRankings(null,
                                       stores as D2Store[]);
       }
     },
 
-    updateCachedUserRankings(item: DimItem, userReview) {
+    updateCachedUserRankings(item: DimItem, userReview: WorkingD1Rating | WorkingD2Rating) {
       if (item.isDestiny1()) {
         _reviewDataCache.addUserReviewData(item,
-                                           userReview);
+                                           userReview as WorkingD1Rating);
```

Comment:
You can implement a type predicate for this that lets you avoid doing "as": http://www.typescriptlang.org/docs/handbook/advanced-types.html